[SFML] build failure due to vcpkg applied patch

### Operating system

MacOS and Linux

### Compiler

GCC, Clang, AppleClang

### Steps to reproduce the behavior

```Shell
The easiest way is to test the original PR since my current PR is trying to update Catch2 to version 3 
Original PR : https://github.com/mod-cpp/ms-pacman/pull/21/checks

git clone https://github.com/mod-cpp/ms-pacman.git
cd ms-pacman
git checkout update_vcpkg #branch from the PR
cmake --preset linux-gcc -DCMAKE_BUILD_TYPE=Debug # configure WILL FAIL
cmake --preset linux-gcc -DCMAKE_BUILD_TYPE=Debug # configure WILL PASS
cmake --build --preset linux-gcc-build --config Debug # build
ctest --preset linux-gcc-test -C Debug # run tests
```


### Failure logs

Linux

```
CMake Error at build/linux-clang/vcpkg_installed/x64-linux/share/sfml/SFMLConfig.cmake:144 (message):
  SFML found but some of its dependencies are missing ( OpenGL FreeType)
Call Stack (most recent call first):
  build/linux-clang/_deps/vcpkg-src/scripts/buildsystems/vcpkg.cmake:852 (_find_package)
  lib/CMakeLists.txt:1 (find_package)

CMake Error at build/linux-clang/_deps/vcpkg-src/scripts/buildsystems/vcpkg.cmake:852 (_find_package):
  Found package configuration file:

    /home/patricia-gallardo/Code/ms-pacman/build/linux-clang/vcpkg_installed/x64-linux/share/sfml/SFMLConfig.cmake

  but it set SFML_FOUND to FALSE so package "SFML" is considered to be NOT
  FOUND.
Call Stack (most recent call first):
  lib/CMakeLists.txt:1 (find_package)
```

MacOS
```
CMake Error at build/osx/vcpkg_installed/x64-osx/share/sfml/SFMLConfig.cmake:144 (message):
  SFML found but some of its dependencies are missing ( OpenGL FreeType)
Call Stack (most recent call first):
  build/osx/_deps/vcpkg-src/scripts/buildsystems/vcpkg.cmake:852 (_find_package)
  lib/CMakeLists.txt:1 (find_package)


CMake Error at build/osx/_deps/vcpkg-src/scripts/buildsystems/vcpkg.cmake:852 (_find_package):
  Found package configuration file:

    /Users/patricia-gallardo/Code/ms-pacman/build/osx/vcpkg_installed/x64-osx/share/sfml/SFMLConfig.cmake

  but it set SFML_FOUND to FALSE so package "SFML" is considered to be NOT
  FOUND.
Call Stack (most recent call first):
  lib/CMakeLists.txt:1 (find_package)
```

### Additional context

We are trying to update to a new vcpkg version and we're having trouble with SFML, it gives an error but if we run CMake again it's fine.

This patch seems like it might be related:
https://github.com/microsoft/vcpkg/blob/master/ports/sfml/fix-dependencies.patch

According to @cor3ntin's experiments the patch is what is causing the problem. Not applying the patch fixes the issue.

We first saw this back in January when we tried to update vcpkg https://github.com/mod-cpp/ms-pacman/pull/21

At the time we just put it on the backburner and didn't report as we should have.

Now I tried to update again in https://github.com/mod-cpp/ms-pacman/pull/25

And I am "solving" it there by running CMake config twice if the first one fails:

https://github.com/mod-cpp/ms-pacman/pull/25/commits/84cb7cfc4eded4ab2d1f27393928c88b41cbfe1c

Specifically like this in the GitHub workflow yaml file:

`cmake --preset ${{ matrix.configurations.config-preset }} -DCMAKE_BUILD_TYPE=${{ matrix.cmake-build-type }} || 
cmake --preset ${{ matrix.configurations.config-preset }} -DCMAKE_BUILD_TYPE=${{ matrix.cmake-build-type }}`

Typical failing build ([new PR 25](https://github.com/mod-cpp/ms-pacman/pull/25)): https://github.com/mod-cpp/ms-pacman/actions/runs/5255810326
Typical failing build ([old PR 21 ](https://github.com/mod-cpp/ms-pacman/pull/21)- easier to repro currently): https://github.com/mod-cpp/ms-pacman/actions/runs/5256326977

Windows builds, everything else fails.

--- a/ports/sfml/fix-dependencies.patch
+++ b/ports/sfml/fix-dependencies.patch
@@ -1,5 +1,5 @@
 diff --git a/cmake/SFMLConfig.cmake.in b/cmake/SFMLConfig.cmake.in
-index ce81953..bd2d330 100644
+index ce81953..c8dd971 100644
 --- a/cmake/SFMLConfig.cmake.in
 +++ b/cmake/SFMLConfig.cmake.in
 @@ -1,3 +1,8 @@
@@ -12,10 +12,22 @@ index ce81953..bd2d330 100644
  # ------------------------------------
  #
 diff --git a/cmake/SFMLConfigDependencies.cmake.in b/cmake/SFMLConfigDependencies.cmake.in
-index 1028110..927d1a4 100644
+index 1028110..76fc650 100644
 --- a/cmake/SFMLConfigDependencies.cmake.in
 +++ b/cmake/SFMLConfigDependencies.cmake.in
-@@ -56,6 +56,9 @@ if(SFML_STATIC_LIBRARIES)
+@@ -31,9 +31,11 @@ if(SFML_STATIC_LIBRARIES)
+         endif()
+ 
+         # No lookup in environment variables (PATH on Windows), as they may contain wrong library versions
++        if(NOT ${THIS_FRIENDLY_NAME}_LIB)
+         find_library(${THIS_FRIENDLY_NAME}_LIB NAMES ${THIS_SEARCH_NAMES}
+                      PATHS ${FIND_SFML_PATHS} PATH_SUFFIXES lib NO_SYSTEM_ENVIRONMENT_PATH)
+         mark_as_advanced(${THIS_FRIENDLY_NAME}_LIB)
++        endif()
+         if(${THIS_FRIENDLY_NAME}_LIB)
+             set_property(TARGET ${THIS_TARGET} APPEND PROPERTY INTERFACE_LINK_LIBRARIES "${${THIS_FRIENDLY_NAME}_LIB}")
+         else()
+@@ -56,6 +58,9 @@ if(SFML_STATIC_LIBRARIES)
          if (FIND_SFML_OS_WINDOWS)
              set_property(TARGET OpenGL APPEND PROPERTY INTERFACE_LINK_LIBRARIES "OpenGL32")
          elseif(NOT FIND_SFML_OS_IOS)
@@ -25,7 +37,7 @@ index 1028110..927d1a4 100644
              sfml_bind_dependency(TARGET OpenGL FRIENDLY_NAME "OpenGL" SEARCH_NAMES "OpenGL" "GL")
          endif()
      endif()
-@@ -63,6 +66,8 @@ if(SFML_STATIC_LIBRARIES)
+@@ -63,6 +68,8 @@ if(SFML_STATIC_LIBRARIES)
      # sfml-graphics
      list(FIND SFML_FIND_COMPONENTS "graphics" FIND_SFML_GRAPHICS_COMPONENT_INDEX)
      if(FIND_SFML_GRAPHICS_COMPONENT_INDEX GREATER -1)
@@ -35,7 +47,7 @@ index 1028110..927d1a4 100644
      endif()
  
 diff --git a/src/SFML/Graphics/CMakeLists.txt b/src/SFML/Graphics/CMakeLists.txt
-index 883c758..402efbe 100644
+index 883c758..b59e659 100644
 --- a/src/SFML/Graphics/CMakeLists.txt
 +++ b/src/SFML/Graphics/CMakeLists.txt
 @@ -97,7 +97,8 @@ sfml_add_library(sfml-graphics
@@ -48,7 +60,7 @@ index 883c758..402efbe 100644
  
  # let CMake know about our additional graphics libraries paths
  if(SFML_OS_WINDOWS)
-@@ -134,7 +134,8 @@ if(SFML_OS_ANDROID)
+@@ -134,7 +135,8 @@ if(SFML_OS_ANDROID)
      target_link_libraries(sfml-graphics PRIVATE z EGL GLESv1_CM)
  endif()
  
@@ -59,7 +71,7 @@ index 883c758..402efbe 100644
  
  # add preprocessor symbols
 diff --git a/src/SFML/Window/CMakeLists.txt b/src/SFML/Window/CMakeLists.txt
-index 98ea439..0f1fb53 100644
+index 98ea439..acb8d61 100644
 --- a/src/SFML/Window/CMakeLists.txt
 +++ b/src/SFML/Window/CMakeLists.txt
 @@ -254,7 +254,9 @@ if(SFML_OPENGL_ES)

--- a/ports/sfml/vcpkg.json
+++ b/ports/sfml/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "sfml",
   "version": "2.5.1",
-  "port-version": 15,
+  "port-version": 16,
   "description": "Simple and fast multimedia library",
   "homepage": "https://github.com/sfml/sfml",
   "license": "Zlib",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -7422,7 +7422,7 @@
     },
     "sfml": {
       "baseline": "2.5.1",
-      "port-version": 15
+      "port-version": 16
     },
     "sfsexp": {
       "baseline": "1.3.1",

--- a/versions/s-/sfml.json
+++ b/versions/s-/sfml.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d9e87b4e23cdcec8d249517f14a69d8e453d3418",
+      "version": "2.5.1",
+      "port-version": 16
+    },
+    {
       "git-tree": "477c029a9588d5f24c19fd0b6dbc82813fdc2150",
       "version": "2.5.1",
       "port-version": 15

